### DISCUSSION TITLE
How can an image title be added into the editor's json during upload?

### DISCUSSION THREAD

**Discussion by meshackm** [steven-tey/novel#366] [Q&A] [answered]
Hey devs, am using Novel's editor for a project with R2 as the image storage. 
I am using signedURLs, thus storing image urls in the database is not viable.
I want to store the image key in the title and use it to generate a signedURL upon request.
Is there a way to go about this?

**Comment by meshackm** [comment, marked as answer]
I ended up saving the image src and image key in localstorage then when getting the editor json onUpdate, a loop checks for image src and compares it to the one in local storage. if they are the same then the title is assigned to the value of image key. Lastly the key and it's value are removed from localStorage after uploading editor json to database. This only happens once during image upload.

**Comment by ankuragrwl** [comment]
Hi @meshackm,

Could you please share more detail on this? I want to capture Image Caption and Alt Text while uploading the image in the editor and also assign these values to JSON.
Thanks
